CRM/Price/BAO/LineItem.php

```diff
@@ -605,4 +605,266 @@ public static function getLineItemArray(&$params, $entityId = NULL, $entityTable
     }
   }
 
+  /**
+   * Function to update related contribution of a entity and
+   *  add/update/cancel financial records
+   *
+   * @param array $params
+   * @param int $entityID
+   * @param int $entity
+   * @param int $contributionId
+   * @param $feeBlock
+   * @param array $lineItems
+   * @param $paidAmount
+   *
+   */
+  public static function changeFeeSelections(
```

Comment:
This function is around 240 lines. Could we split it up for readability purposes into a few smaller helper functions that are protected and named beginning with _ ?